tests/lib/rules/no-warning-comments.js

```diff
@@ -387,6 +389,45 @@ ruleTester.run("no-warning-comments", rule, {
                     }
                 }
             ]
+        },
+        {
+            code: "/*\nTODO undecorated multi-line block comment (start)\n*/",
+            options: [{ terms: ["todo"], location: "start" }],
+            errors: [
+                {
+                    messageId: "unexpectedComment",
+                    data: {
+                        matchedTerm: "todo",
+                        comment: "TODO undecorated multi-line block..."
+                    }
+                }
+            ]
+        },
+        {
+            code: "///// TODO decorated single-line comment with decoration array \n /////",
+            options: [{ terms: ["todo"], location: "start", decoration: ["*", "/"] }],
+            errors: [
+                {
+                    messageId: "unexpectedComment",
+                    data: {
+                        matchedTerm: "todo",
+                        comment: "/// TODO decorated single-line comment..."
+                    }
+                }
+            ]
+        },
+        {
+            code: "///*/*/ TODO decorated single-line comment with multiple decoration characters (start) \n /////",
+            options: [{ terms: ["todo"], location: "start", decoration: ["*", "/"] }],
+            errors: [
+                {
+                    messageId: "unexpectedComment",
+                    data: {
+                        matchedTerm: "todo",
+                        comment: "/*/*/ TODO decorated single-line comment..."
+                    }
+                }
+            ]
```

Comment:
Can you also add a test case where the term starts with decoration characters, for example:

```js
{
    code: "//**TODO",
    options: [{ terms: ["*todo"], location: "start", decoration: ["*"] }],
    errors: [
        {
            messageId: "unexpectedComment",
            data: {
                matchedTerm: "*todo",
                comment: "**TODO"
            }
        }
    ]
}
```